src/process_watch.js

```diff
@@ -0,0 +1,369 @@
+
+  })
+  if (processWarnList.some(pidStored)) {
+    processObject.lastDetectTime = currentTime;
+    //if exists in storage
+    // update; cpu, memory, lastDetectTime, 
+
+    //element.lastDetectTime <= (currentTime - reportInterval)
+    index = _.findIndex(processWarnList, function (element) {
+      return element.pid == processObject.pid
+    })
+    /* 
+     * found time, Show how long a ps has been on list
+     * age should be difference between found and current time 
+     */
+    processObject.firstDetectTime = processWarnList[index].firstDetectTime;
+    // reportTime, Show how long since last report
+    processObject.reportTime = processWarnList[index].reportTime;
+    // If a new process, that is repeat offender, allow reporting it
+    // Also a little over 15 min old 
+    processObject.report = ((processObject.reportTime == undefined) && (
+      processObject.firstDetectTime <= (currentTime - 15 * 60 * 1000)));
+    //overwrite previous
+    processWarnList.splice(index, 1, processObject);
+  } else {
+    processObject.lastDetectTime = currentTime;
+    processObject.firstDetectTime = currentTime;
+    processObject.report = false;
+    processWarnList.push(processObject);
+  }
+}
+
+module.exports = {
+  /**
+   * init
+   * @param {ps} current-processes npm module
+   * @param {pid} pidusage npm module
+   * @param {steam} output stream for reports. Usually Slack. STDout for testing
+   * bring in the process watching modules
+   * AND the stream for output to Slack
+   */
+  init: function (ps, pid, stream) {
+    psLookup = ps;
+    pidusage = pid;
+    currStream = stream;
+  },
+
+  /**
+   * checkProcess
+   * @param {currentTime} used track in lists when 'new' and when 'updated'
+   * On first new run we check cpu type
+   * 
+   * new cpuAddLevel processes are added to Watch list
+   * If a  PID 'match' is in Watch already, process is added to Warn list
+   * 
+   * If tracked process is above cpuAddLevel, it's 'last detected' timestamp is updated
+   * If a process is below cpuAddLevel, it can eventually age out of the list
+   * 
+   * 'cleanList' is then run with 'processes,' a current snapshot of all processes.
+   */
+  checkProcess: function checkProcess(currentTime = new Date().getTime()) {
+    // stops function from crashing server if init hasn't been run
+    if (!psLookup) {
+      console.log("process_watch not initialized");
+      return null;
+    }
+    cpuTypeChecked = false;
+    psLookup.get(function (err, processes) {
+
+      var sorted = _.sortBy(processes, "cpu");
+      let top5 = sorted.reverse().splice(0, 5);
+      top5.forEach(element => {
+        // tests need to match what OS returns for cpu
+        // 25 or .25
+        if (!cpuTypeChecked) {
+          updateCPUTestValues(element.cpu);
+        }
+        //  This allows testing enviroment to pass in times,
+        //  normally current-processes doesn't pass a timestamp
+        currentTime = ((typeof (element.timestamp) !==
```

Comment:
if your doing this for the testing process ... doesn't the definition of `checkProcess( currentTime )`  allow you to pass in the current time for testing?

Reply:
... yes, It does. Guess I wanted to be REALLY sure current time got set